src/Akihabara.Tests/Framework/ImageFormat/ImageFrameTest.cs

```diff
@@ -17,51 +17,54 @@ public class ImageFrameTest
         [Test, SignalAbort]
         public void Ctor_ShouldInstantiateImageFrame_When_CalledWithNoArguments()
         {
-            var imageFrame = new ImageFrame();
-
-            Assert.AreEqual(imageFrame.Format(), ImageFormat.Format.Unknown);
-            Assert.AreEqual(imageFrame.Width(), 0);
-            Assert.AreEqual(imageFrame.Height(), 0);
-            Assert.Throws<FormatException>(() => { imageFrame.ChannelSize(); });
-            Assert.Throws<FormatException>(() => { imageFrame.NumberOfChannels(); });
-            Assert.Throws<FormatException>(() => { imageFrame.ByteDepth(); });
-            Assert.AreEqual(imageFrame.WidthStep(), 0);
-            Assert.AreEqual(imageFrame.PixelDataSize(), 0);
-            Assert.Throws<FormatException>(() => { imageFrame.PixelDataSizeStoredContiguously(); });
-            Assert.True(imageFrame.IsEmpty());
-            Assert.False(imageFrame.IsContiguous());
-            Assert.False(imageFrame.IsAligned(16));
-            Assert.AreEqual(imageFrame.MutablePixelData(), IntPtr.Zero);
+            using (var imageFrame = new ImageFrame())
+            {
+                Assert.AreEqual(imageFrame.Format(), ImageFormat.Format.Unknown);
+                Assert.AreEqual(imageFrame.Width(), 0);
+                Assert.AreEqual(imageFrame.Height(), 0);
+                Assert.Throws<FormatException>(() => { imageFrame.ChannelSize(); });
+                Assert.Throws<FormatException>(() => { imageFrame.NumberOfChannels(); });
+                Assert.Throws<FormatException>(() => { imageFrame.ByteDepth(); });
+                Assert.AreEqual(imageFrame.WidthStep(), 0);
+                Assert.AreEqual(imageFrame.PixelDataSize(), 0);
+                Assert.Throws<FormatException>(() => { imageFrame.PixelDataSizeStoredContiguously(); });
+                Assert.True(imageFrame.IsEmpty());
+                Assert.False(imageFrame.IsContiguous());
+                Assert.False(imageFrame.IsAligned(16));
+                Assert.AreEqual(imageFrame.MutablePixelData(), IntPtr.Zero);
+            }
```

Comment:
Oh, I see. So using a `using` block here lets us tell GC exactly when to dispose of the `ImageFrame`? And so no modification inside of `ImageFrame.cs` is required?

Reply:
```
The using statement ensures that Dispose (or DisposeAsync) is called even if an exception occurs within the using block. You can achieve the same result by putting the object inside a try block and then calling Dispose (or DisposeAsync) in a finally block; in fact, this is how the using statement is translated by the compiler.
```
Interesting... Though the last problem is that the bugs that arise when we don't do that are really hard to figure out and they're at runtime. Is it OK to leave it this way so that you always have to use a `using` block or explicitly dispose in a `finally`? Is there no solution to make this problem explicit to the user in types or something?

Reply:
> Is it OK to leave it this way so that you always have to use a using block or explicitly dispose in a finally? Is there no solution to make this problem explicit to the user in types or something?

If you don't need `AOT` (it's unavoidable in Unity), then I recommend you implement `ReleasePixelData` as an instance method.
Then this lifetime problem won't occur. In this case, note that you need to pin the address of the function pointer (cf. https://docs.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.gchandle.alloc?view=net-5.0).

By the way, I think using `using` (not a pun) is [a recommended way](https://docs.microsoft.com/en-us/dotnet/api/system.idisposable?view=net-5.0#using-an-object-that-implements-idisposable) to use `IDisposable` object, so it's not too weird.
If you don't dispose such a object at a proper moment, memory will leak because GC does not know how much memories are actually used (we can tell GC that, but it's tedious).

Reply:
I see. If we want to merge our works at one point, we'll have to take `AOT` into account.
Using `using` seems to be the way though, so I'll leave it at that.

Reply:
We plan to put some of the Unity specific stuff in as per #28 by Encore so we'll have to look into this further. You can continue the discussion there.